pkg/types/types.go

```diff
@@ -154,35 +154,65 @@ func (m *ComMatrix) WriteMatrixToFileByType(utilsHelpers utils.UtilsInterface, f
 	return nil
 }
 
-// GenerateMatrixDiff generates the diff between mat1 to mat2.
-func (m *ComMatrix) GenerateMatrixDiff(mat2 *ComMatrix) (string, error) {
-	colNames, err := getComMatrixHeadersByFormat(FormatCSV)
-	if err != nil {
-		return "", fmt.Errorf("error getting commatrix CSV tags: %v", err)
-	}
-
-	diff := colNames + "\n"
+// markDiffBetweenMatrices map the cd's string of the matrices to ints in the following way:
+// cd which m contains but other doesn't --> 1
+// cd both m and other contains --> 0
+// cd which other doesn't contain but m does --> -1.
+func (m *ComMatrix) markDiffBetweenMatrices(other *ComMatrix) map[string]int {
+	mapComDetailToSign := make(map[string]int)
 
 	for _, cd := range m.Matrix {
-		if mat2.Contains(cd) {
-			diff += fmt.Sprintf("%s\n", cd)
-			continue
+		if other.Contains(cd) {
+			mapComDetailToSign[cd.String()] = 0
+		} else {
+			// m contains cd but other doesn't
+			mapComDetailToSign[cd.String()] = 1
 		}
-
-		// add "+" before cd's mat1 contains but mat2 doesn't
-		diff += fmt.Sprintf("+ %s\n", cd)
 	}
 
-	for _, cd := range mat2.Matrix {
+	for _, cd := range other.Matrix {
 		// Skip "rpc.statd" ports, these are randomly open ports on the node,
 		// no need to mention them in the matrix diff
 		if cd.Service == "rpc.statd" {
 			continue
 		}
 
 		if !m.Contains(cd) {
-			// add "-" before cd's mat1 doesn't contain but mat2 does
+			// m doesn't contain cd but other does
+			mapComDetailToSign[cd.String()] = -1
+		}
+	}
+
+	return mapComDetailToSign
+}
+
+// GenerateMatrixDiff generates the diff between mat1 to mat2.
+func (m *ComMatrix) GenerateDiff(other *ComMatrix) (string, error) {
+	combinedComDetails := append([]ComDetails{}, m.Matrix...)
```

Comment:
combinedComDetails := append(m.Matrix, other.Matrix...)

Reply:
I'm getting a linting error when doing so.

Reply:
`appendAssign: append result not assigned to the same slice (gocritic)`

Reply:
ok so you can do
```
combindComDetails := m.Matrix
combinedComDetails = append(combinedComDetails, other.Matrix...)
```

Reply:
That way it shallow copies the matrix since it is a slice, I think it might be better to deep copy as I did so the original won't change as we iterate through it for mapping the sign. what do you think?

Reply:
I think for our goal it's sufficient but let's keep it as it is